### PR TITLE
feat(Toc): add component

### PR DESCRIPTION
**Pull request by chelentos**

### CODE REVIEW

File: src/components/Toc/Toc.scss [hunk 1]
```diff
@@ -0,0 +1,17 @@
+@use '../variables';
+
+$block: '.#{variables.$ns}toc';
```
Comment (amje) [inline src/components/Toc/Toc.scss:3]: `$ns` is deprecated. Use `$ns-new` instead

File: src/components/Toc/Toc.scss [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+#{$block} {
+    &__title {
+        font-size: var(--g-text-body-2-font-size);
+        font-weight: 500;
```
Comment (amje) [inline src/components/Toc/Toc.scss:8]: Should use one of typography [mixins](https://github.com/gravity-ui/uikit/blob/13ed356b1f8eaf82816f0f6e0ec3f6437b364c41/styles/mixins.scss#L149), font properties can be configured in theme and must not be hardcoded

File: src/components/Toc/TocItem/TocItem.scss [hunk 3]
```diff
@@ -0,0 +1,41 @@
+@use '../../variables';
+
+$block: '.#{variables.$ns}toc-item';
```
Comment (amje) [inline src/components/Toc/TocItem/TocItem.scss:3]: `$ns-new`

File: src/components/Toc/TocItem/TocItem.tsx [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+    const {childItem = false, active = false, onClick, title, value} = props;
+
+    const handleClick = () => onClick(value);
+
+    const {onKeyDown} = useActionHandlers(handleClick);
+
+    return (
+        <div className={b('section', {child: childItem, active})}>
+            <div
+                role="radio"
```
Comment (amje) [inline src/components/Toc/TocItem/TocItem.tsx:27]: Why `radio`? For a11y would be better to wrap `ul/li` into `nav`

File: src/components/Toc/types.ts [hunk 5]
```diff
@@ -0,0 +1,5 @@
+export interface TocItem {
+    value: string;
+    title?: React.ReactNode;
+    selector?: string;
```
Comment (amje) [inline src/components/Toc/types.ts:4]: Can be deleted I guess?

File: src/components/Toc/types.ts [hunk 6]
```diff
@@ -0,0 +1,5 @@
+export interface TocItem {
+    value: string;
+    title?: React.ReactNode;
```
Comment (amje) [inline src/components/Toc/types.ts:3]: Let's call it `content`?

**Review by amje — CHANGES_REQUESTED** [review]

**Comment by amje** [comment]
Also I'd use `<a>` in item for better a11y

File: src/components/Toc/TocItem/TocItem.tsx [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+export interface TocItemProps extends TocItemType {
+    childItem?: boolean;
+    active?: boolean;
+    onClick: (value: string) => void;
+}
+
+export const TocItem = (props: TocItemProps) => {
+    const {childItem = false, active = false, onClick, title, value} = props;
+
+    const handleClick = () => onClick(value);
```
Comment (imechoim) [inline src/components/Toc/TocItem/TocItem.tsx:20]: Please, Wrap it in React.useCallback.

**Review by amje — CHANGES_REQUESTED** [review]

File: src/components/Toc/types.ts [hunk 8]
```diff
@@ -0,0 +1,9 @@
+export interface TocItem {
+    value?: string;
+    content?: React.ReactNode;
+    href?: string;
+}
+
+export type TocItems = (TocItem & {
+    items?: TocItem[];
+})[];
```
[the remark this answers is not in the document]
Comment (chelentos) [reply src/components/Toc/types.ts:9]: Because we only have 2 levels of hierarchy in Toc

[the remark this answers is not in the document]
Comment (amje) [reply src/components/Toc/types.ts:9]: It's ok and maybe we'll support any level in the future.

File: src/components/Toc/TocItem/TocItem.scss [hunk 9]
```diff
@@ -0,0 +1,41 @@
...
+            padding: 6px 6px 6px 12px;
+            min-height: 28px;
+
+            color: var(--g-color-text-secondary);
+            border-left: 2px solid transparent;
+            text-decoration: none;
+
+            &:hover {
+                color: var(--g-color-text-complementary);
+            }
```
[the remark this answers is not in the document]
Comment (chelentos) [reply src/components/Toc/TocItem/TocItem.scss:40]: can't use general mixin here because of paddings and component specifics

**Review by amje — APPROVED** [review]

**Pull request merged**